deadlock-dotnet-sdk/Domain/SafeHandleEx.cs

```diff
@@ -0,0 +1,273 @@
+
+            /** Get Process's name */
+            if (!string.IsNullOrWhiteSpace(ProcessMainModulePath))
+            {
+                ProcessName = Path.GetFileNameWithoutExtension(ProcessMainModulePath);
+            }
+
+            /** Get process's possibly-overwritten command line from the PEB struct in its memory space */
+            GetProcessCommandLine(hProcess);
+        }
+        catch (Exception e)
+        {
+            ExceptionLog.Add(e);
+        }
+    }
+
+    internal NativeMethods.SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX SysHandleEx { get; }
+
+    public unsafe void* Object => SysHandleEx.Object;
+    /// <summary>
+    /// cast to uint
+    /// </summary>
+    public HANDLE ProcessId => SysHandleEx.UniqueProcessId;
+    public HANDLE HandleValue => SysHandleEx.HandleValue;
+    public ushort CreatorBackTraceIndex => SysHandleEx.CreatorBackTraceIndex;
+    /// <inheritdoc cref="SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX.GrantedAccess"/>
+    public ACCESS_MASK GrantedAccess => SysHandleEx.GrantedAccess;
+    public ushort ObjectTypeIndex => SysHandleEx.ObjectTypeIndex;
+    /// <inheritdoc cref="SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX.HandleAttributes"/>
+    public uint HandleAttributes => SysHandleEx.HandleAttributes;
+
+    /// <summary>
+    /// The Type of the object as a string.
+    /// </summary>
+    /// <value></value>
+    public string? HandleObjectType { get; private set; }
+
+    public string? ProcessCommandLine { get; private set; }
+    public string? ProcessMainModulePath { get; private set; }
+    public string? ProcessName { get; private set; }
+
+    /// <summary>
+    /// A list of exceptions thrown by constructors and other methods of this class.<br/>
+    /// Intended to explain why the process command line, main module path, and name are unavailable.
+    /// </summary>
+    /// <remarks>Use List's methods (e.g. Add) to modify this list.</remarks>
+    public List<Exception> ExceptionLog { get; } = new();
+
+    /// <summary>
+    /// Release the system handle.<br/>
+    /// ! WARNING !<br/>
+    /// If the handle or a duplicate is in use by a driver or other kernel-level software, a function that accesses the now-invalid handle will cause a stopcode (AKA Blue Screen Of D).
+    /// </summary>
+    /// <remarks>
+    /// See Raymond Chen's devblog article 
+    /// <see href="https://devblogs.microsoft.com/oldnewthing/20070829-00/?p=25363">"Kernel handles are not reference-counted"</see>.
+    /// </remarks>
+    public void UnlockSystemHandle()
+    {
+        HANDLE rawHProcess;
+        SafeProcessHandle? hProcess = null;
+        try
+        {
+            if ((rawHProcess = OpenProcess(
+                PROCESS_ACCESS_RIGHTS.PROCESS_DUP_HANDLE,
+                true,
+                (uint)ProcessId)
+                ).IsNull)
+            {
+                throw new Win32Exception($"Failed to open process with id {(int)ProcessId} to duplicate and close object handle.");
```

Comment:
@BinToss You can get the last Win32 error from this function here too:
https://learn.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror